pkg/kube/pod.go

```diff
@@ -295,11 +295,17 @@ func DeletePod(ctx context.Context, cli kubernetes.Interface, pod *corev1.Pod) e
 	return nil
 }
 
-func StreamPodLogs(ctx context.Context, cli kubernetes.Interface, namespace, podName, containerName string) (io.ReadCloser, error) {
+func StreamPodLogs(ctx context.Context, cli kubernetes.Interface, namespace, podName, containerName string, sinceNow bool) (io.ReadCloser, error) {
 	plo := &corev1.PodLogOptions{
 		Follow:    true,
 		Container: containerName,
 	}
+	if sinceNow {
```

Comment:
Should it be sinceNow? Maybe we can somehow figure out the time when previous log ended?
I'm thinking if we can lose some logs this way. Probably having duplicate logs and deduplicating on log parsing is better than potentially losing some. WDYT?

Reply:
Even though this solution will work in some (maybe even in most of the cases), it would be better to have something more robust.

The `io.Reader` returned by `StreamPodLogs` could be wrapped, and the last X chunks of data read, could be stored in buffer, with timestamp when it was read. 
Then, when reader will be closed with EOF, streaming could be restarted since last seen data (maybe even 10 seconds before). Then everything (including seen data) need to be skipped.
In such case we will have confidence that we won't skip any output line.

Reply:
Sorry for closing, I had a misunderstanding in my organization. I opened a new [PR](https://github.com/kanisterio/kanister/pull/2972) for follow-up action.

In the new PR, I have already taken this comment into account. We store the time of the last successful read, and after EOF, we request logs from that time. During tests, it shows that kubelet will always return logs starting with the last read line (even after a few hours). This always leads to duplication, never a loss of information.

Reply:
#2972